src/Provider/DecoratingCacheSettingsProvider.php

```diff
@@ -228,66 +228,101 @@ private function clearIfNeeded(): void
         $this->cache->save($lastCheck);
     }
 
-    private function warmup(array $domainNames, array $settingNames = null): void
+    private function warmupByDomainNames(array $domainNames): void
     {
         foreach ($domainNames as $domainName) {
             if ($this->isDomainSettingsWarm($domainName)) {
                 continue;
             }
 
-            if (null === $settingNames) {
-                $lock = $this->lockFactory->createLock(__FUNCTION__.$domainName);
-                if (!$lock->acquire()) {
-                    usleep(self::LOCK_RETRY_INTERVAL_MS);
-                    $this->warmup($domainNames, $settingNames);
+            $lock = $this->lockFactory->createLock(__FUNCTION__.$domainName);
+            if (!$lock->acquire()) {
+                usleep(self::LOCK_RETRY_INTERVAL_MS);
+                $this->warmupByDomainNames($domainNames);
 
-                    return;
-                }
+                return;
+            }
 
-                try {
-                    $this->warmupDomainSettings($domainName);
-                } finally {
-                    $lock->release();
-                }
-            } else {
-                foreach ($settingNames as $settingName) {
-                    if ($this->isSettingWarm($domainName, $settingName)) {
-                        continue;
-                    }
-
-                    $lock = $this->lockFactory->createLock(__FUNCTION__.$domainName.$settingName);
-                    if (!$lock->acquire()) {
-                        usleep(self::LOCK_RETRY_INTERVAL_MS);
-                        $this->warmup($domainNames, $settingNames);
-
-                        return;
-                    }
-
-                    try {
-                        $this->warmupSingleSetting($domainName, $settingName);
-                    } finally {
-                        $lock->release();
-                    }
-                }
+            try {
+                $this->warmupDomainSettings($domainName);
```

Comment:
Is it worth doing similar optimisation here? Now Provider is called for each domain separately, but might be changed to do single call for all domains in a similar manner.